agent/engine/task_manager.go

```diff
@@ -73,6 +81,17 @@ type containerTransition struct {
 	reason         error
 }
 
+// resourceTransition defines the struct for a resource to transition.
+type resourceTransition struct {
+	// nextState represents the next known status that the resource can move to
+	nextState taskresource.ResourceStatus
+	// status is the string value of nextState
+	status string
```

Comment:
Can we get rid of this field, as it's duplicate with `nextState`?

Reply:
We use this field [here](https://github.com/aws/amazon-ecs-agent/pull/1332/files#diff-20346f7af6202828de353f0137b6e43aR859) and to track the map of transitions [here](https://github.com/aws/amazon-ecs-agent/pull/1332/files#diff-20346f7af6202828de353f0137b6e43aR741). Without the resource object, we cannot get the status string, hence added it as part of `resourceTransition`.